flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/HeapKeyedStateBackend.java

```diff
@@ -446,8 +485,10 @@ public String toString() {
 	@Override
 	public int numStateEntries() {
 		int sum = 0;
-		for (StateTable<K, ?, ?> stateTable : stateTables.values()) {
-			sum += stateTable.size();
+		for (StateSnapshotRestore stateTable : registeredStates.values()) {
+			if (stateTable instanceof StateTable) {
+				sum += ((StateTable<?, ?, ?>) stateTable).size();
+			}
```

Comment:
Why does the timers don't count for the total number of state entries?

Reply:
This method is only used for some tests, and to be on the safe side it probably only expected to count the keyed state and not some timers.